mobile/src/main/java/ch/epfl/sdp/mobile/state/game/delegating/DelegatingSpeechRecognizerState.kt

```diff
@@ -59,17 +61,19 @@ constructor(
             when (val speech = facade.recognize()) {
               // TODO : Display an appropriate message, otherwise act on the board.
               SpeechFacade.RecognitionResult.Failure.Internal ->
-                  snackbarHostState.showSnackbar("Internal failure")
+                  currentError = SpeechRecognizerError.InternalError
               is SpeechFacade.RecognitionResult.Success -> {
 
-                val parsedValue = VoiceInput.parseInput(speech.results)
-                snackbarHostState.showSnackbar(parsedValue.toString())
-
-                // TODO(Chau) : Do something more interesting
-                Position.all()
-                    .flatMap { delegate.game.actions(it) }
-                    .onEach { delegate.tryPerformAction(it) }
-                    .firstOrNull()
+                // Parsed the input
+                val parsedAction = VoiceInput.parseInput(speech.results)
+                if (parsedAction != null) {
+                  // Try to perform the action
+                  val isSuccessful = delegate.tryPerformAction(parsedAction)
+                  if (!isSuccessful) currentError = SpeechRecognizerError.IllegalAction
+                  // If the action is illegal show in the snackbar
+                } else { // If cannot be parsed show error message
+                  currentError = SpeechRecognizerError.UnknownCommand
+                }
```

Comment:
This code could be simplified to 
```
currentError = parsedAction?.let {
                   val isSuccessful = delegate.tryPerformAction(it)
                   if (!isSuccessful) SpeechRecognizerError.IllegalAction else currentError}
                   ?: SpeechRecognizerError.UnknownCommand
```

Reply:
I don't really like the else part (`else currentError`). For me, it's not logical to assign the error variable if there is no error. In addition to this, this is quite ambiguous and error-prone to assign the same variable.

I suggest leaving it as it is

Reply:
Ok 👍